src/eval_context.rs

```diff
@@ -1353,7 +1347,12 @@ impl<'a, 'tcx> EvalContext<'a, 'tcx> {
                     I128 => 16,
                     Is => self.memory.pointer_size(),
                 };
-                PrimVal::from_i128(self.memory.read_int(ptr, size)?)
+                // if we cast a ptr to an isize, reading it back into a primval shouldn't panic
```

Comment:
I am not sure I am happy about using the word "cast" here. This is doing a transmute, i.e., it is reading the same bytes at a different type. The rules for the two should be somewhat consistent, but they are clearly not the same right now (as they are implemented in two different files).